graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/SuggestionsResource.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.plugins.views.search.rest;
+
+import com.google.common.collect.ImmutableSet;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import org.apache.shiro.authz.annotation.RequiresAuthentication;
+import org.graylog.plugins.views.search.engine.QueryEngine;
+import org.graylog.plugins.views.search.engine.suggestions.SuggestionRequest;
+import org.graylog.plugins.views.search.engine.suggestions.SuggestionResponse;
+import org.graylog.plugins.views.search.permissions.SearchUser;
+import org.graylog.plugins.views.search.rest.suggestions.SuggestionEntryDTO;
+import org.graylog.plugins.views.search.rest.suggestions.SuggestionsDTO;
+import org.graylog.plugins.views.search.rest.suggestions.SuggestionsErrorDTO;
+import org.graylog.plugins.views.search.rest.suggestions.SuggestionsRequestDTO;
+import org.graylog2.audit.jersey.NoAuditEvent;
+import org.graylog2.plugin.indexer.searches.timeranges.InvalidRangeParametersException;
+import org.graylog2.plugin.indexer.searches.timeranges.RelativeRange;
+import org.graylog2.plugin.rest.PluginRestResource;
+import org.graylog2.shared.rest.resources.RestResource;
+import org.graylog2.shared.security.RestPermissions;
+
+import javax.inject.Inject;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@RequiresAuthentication
+@Api(value = "Search/Suggestions")
+@Path("/search/suggest")
+public class SuggestionsResource extends RestResource implements PluginRestResource {
+
+    private final PermittedStreams permittedStreams;
+    private final QueryEngine queryEngine;
+
+    @Inject
+    public SuggestionsResource(PermittedStreams permittedStreams, QueryEngine queryEngine) {
+        this.permittedStreams = permittedStreams;
+        this.queryEngine = queryEngine;
+    }
+
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation("Validate a search query")
```

Comment:
This is left over from copying the method?

Reply:
Indeed, thanks, fixed!